lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -492,33 +514,43 @@ class ReportUIFeatures {
   /**
    * Opens a new tab to the online viewer and sends the local page's JSON results
    * to the online viewer using postMessage.
-   * @param {LH.Result} reportJson
-   * @param {string} viewerPath
+   * @param {LH.Result} json
+   * @protected
+   */
+  static openTabAndSendJsonReportToViewer(json) {
+    // The popup's window.name is keyed by version+url+fetchTime, so we reuse/select tabs correctly
+    // @ts-ignore - If this is a v2 LHR, use old `generatedTime`.
+    const fallbackFetchTime = /** @type {string} */ (json.generatedTime);
+    const fetchTime = json.fetchTime || fallbackFetchTime;
+    const windowName = `${json.lighthouseVersion}-${json.requestedUrl}-${fetchTime}`;
+    ReportUIFeatures.openTabAndSendData(json, VIEWER_URL, windowName);
+  }
+
+  /**
+   * Opens a new tab to an external page and sends data using postMessage.
+   * @param {Object} data
```

Comment:
https://github.com/GoogleChrome/lighthouse/blob/3c441371977d8f6fcd6ebf6b639ab867aebbae21/lighthouse-viewer/app/src/lighthouse-report-viewer.js#L178

Reply:
Oh, the postMessage part doesn't. ok

Reply:
Why not keep openTabAndSendData generic? the treemap code will eventually send different values than viewer. if not Object it'd eventually need to be typed `{lhr: ...} | LH.Treemap.Options`, which is then just enumerating every callsite's use case.

Reply:
Because in step one we've already tried to pass a malformed payload that doesn't fit that type and I'm sure we'll accidentally do it again someday :)

> which is then just enumerating every callsite's use case.

Why is this a bad thing? If it were truly generic and one could pass in anything they wanted to have it work, I would understand, but when you have to pass in a very particular shape or it's broken, this seems like a good thing to check.